syntax = "proto3";


service UserService {
    // USERS
  rpc CreateRecordUsers (CreateRequestUsers) returns (CreateResponseUsers);
  rpc ReadRecordUsers (ReadRequestUsers) returns (ReadResponseUsers);
  rpc UpdateRecordUsers (UpdateRequestUsers) returns (UpdateResponseUsers);
  rpc DeleteRecordUsers (DeleteRequestUsers) returns (DeleteResponseUsers);
}

message CreateRequestUsers {
  string name = 1;
  bytes key = 2;
  bytes hash = 3;
  string salt = 4;
  int32 status = 5;
  string description = 6;
  string timestamp = 7;
}

message CreateResponseUsers {
  bool success = 1;
  string message = 2;
}


message ReadRequestUsers {
  int64 id = 1;
  string name = 2;
  string key = 3;
  string hash = 4;
  string salt = 5;
  int32 status = 6;
  string description = 7;
  string timestamp = 8;
}

message ReadResponseUsers {
  string name = 1;
  string key = 2;
  string hash = 3;
  string salt = 4;
  int32 status = 5;
  string description = 6;
  string timestamp = 7;
  bool success = 8;
}


message UpdateRequestUsers {
  string name = 1;
  string key = 2;
  string hash = 3;
  string salt = 4;
  int32 status = 5;
  string description = 6;
  string timestamp = 7;
}

message UpdateResponseUsers {
  bool success = 1;
  string message = 2;
}


message DeleteRequestUsers {
  int64 id = 1;
  string name = 2;
  string key = 3;
  string hash = 4;
  string salt = 5;
  int32 status = 6;
  string description = 7;
  string timestamp = 8;
}

message DeleteResponseUsers {
  bool success = 1;
  string message = 2;
}